os/kernel/semaphore/sem_tickwait.c

```diff
@@ -180,7 +180,7 @@ int sem_tickwait(FAR sem_t *sem, systime_t start, uint32_t delay)
 
 	ret = sem_wait(sem);
 	if (ret < 0) {
-		/* Return the errno from sem_wait() */
+		/* Return the errno from sem_wait() or from sem_timeout() */
```

Comment:
Hello, there are 2 scenarios in which task waiting for semaphore will be unblocked as follows
1. If there is timeout happens as part of watchdog timer expiration. See wd_start api above sem_wait. In this case error set as "TIMEOUT"
2. When sem_wait returns.

So ideally we are setting the error no set either by sem_timeout or sem_wait

I hope you agree with me